### PR TITLE
chore: add react hooks linter

### LINKED ISSUE #1739: Fix React rules of hooks issues
Our code breaks the React rule of hook in a few components.
https://reactjs.org/docs/hooks-rules.html
This task is to fix those and add the eslint-plugin-react-hooks to enforce it in the future.
Fixing this will help prevent bugs and React re-render issues.

### PR DESCRIPTION
**Pull request by beguene**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/blockstack/ux/actions/runs/1333445543).

fix existing errors
fix react error
closes #1739 

cc/ @kyranjamie @fbwoolf

### CODE REVIEW

File: src/common/hooks/use-onclickoutside.ts [hunk 1]
```diff
@@ -61,4 +57,6 @@ export default function useOnClickOutside(
       });
     };
   }, [ref, handlerRef, noHandler]);
+
+  if (!isBrowser) return;
```
Comment (kyranjamie) [inline src/common/hooks/use-onclickoutside.ts]: I think this can be removed, was probably copied from explorer

**Review by kyranjamie — APPROVED** [review]
Nice one @beguene—refactor item # 1 ✅

File: src/components/asset-item.tsx [hunk 2]
```diff
@@ -56,7 +55,7 @@ const SubBalance: React.FC<{ amount: string | undefined }> = ({ amount }) =>
   ) : null;
 
 export const AssetItem = memo(
-  forwardRefWithAs(
+  React.forwardRef(
```
Comment (fbwoolf) [inline src/components/asset-item.tsx:58]: Curious, why this change?

**Review by fbwoolf — APPROVED** [review]

File: src/components/asset-item.tsx (hunk 2 above)
Comment (beguene) [reply src/components/asset-item.tsx:58]: this forwardRefWithAs from stacks/ui triggers this 
```
React Hook "usePressable" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function
```
but not the React.forwardRef

Comment (fbwoolf) [reply src/components/asset-item.tsx:58]: Got it, thanks!
